diff --git a/meson.build b/meson.build
index 656ed950a1fb437e05bf9acb3ba6e5b27a719958..6385da1e23a4e3f858a05f129fbcc4cf08e58b4e 100644
--- a/meson.build
+++ b/meson.build
@@ -195,7 +195,7 @@ gettext_args = [ '--msgid-bugs-address=@0@'.format(conf.get('PACKAGE_BUGREPORT')
 
 # Check for libintl.h
 opt_nls = get_option('nls')
-libintl_dep = dependency('intl', required: false, fallback: ['libintl', 'libintl_dep'])
+libintl_dep = dependency('intl', required: opt_nls, fallback: ['libintl', 'libintl_dep'])
 if libintl_dep.found()
   conf.set('ENABLE_NLS', opt_nls.allowed().to_int())
   deps += [libintl_dep]
